src/App.tsx

```diff
@@ -1,156 +1,177 @@
+    setIsLoading(true);
+
+    getTodos(USER_ID).then(
+      fetchedTodos => {
+        setTodos(fetchedTodos as Todo[]);
+        setIsLoading(false);
+      },
+    ).catch(() => setError(TodoErros.ErrorTodo));
+  }, []);
+
+  const preventSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+  };
+
+  const handleImputTodo = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setInput(e.target.value);
+  };
+
+  const handleAddTodo = (
+    e: React.KeyboardEvent<HTMLElement>,
+  ) => {
+    if (input.trim() && e.key === 'Enter') {
+      const newTodo: Omit<Todo, 'id'> = {
+        userId: Number(USER_ID),
+        title: input,
+        completed: false,
+      };
+
+      postTodos(USER_ID, newTodo).then((todo) => {
+        setTodos((prevTodos) => [...prevTodos, todo as Todo]);
+        setInput('');
+
+        if (error && error === TodoErros.Add) {
+          setError('');
+        }
+      }).catch(() => setError(TodoErros.Add));
+    }
+  };
+
+  const handleRemoveTodo = (todoId: number) => {
+    deleteTodos(USER_ID, todoId).then(() => {
+      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+
+      if (error && error === TodoErros.Delete) {
+        setError('');
+      }
+    }).catch(() => setError(TodoErros.Delete));
+  };
+
+  const handleCheckBoxTodo = (todoId: number) => {
+    const curentTodo: Todo | undefined = todos.find(todo => todo.id === todoId);
+
+    if (!curentTodo) {
+      return;
+    }
+
+    patchTodos(todoId, USER_ID, curentTodo).then(() => {
+      setTodos(prevTodos => prevTodos.map(
+        todo => {
+          if (todo.id === todoId) {
+            return {
+              ...todo,
+              completed: !todo.completed,
+            };
+          }
+
+          return todo;
+        },
+      ));
+    });
+  };
+
+  const removeCompletedTodos = () => {
+    setTodos(prevTodos => prevTodos
+      .filter(todo => todo.completed === false));
+  };
+
+  const filteredTodos = filter === FilterTypes.All
+    ? todos
+    : todos.filter((todo) => {
+      if (filter === FilterTypes.Completed) {
+        return todo.completed;
+      }
+
+      return !todo.completed;
+    });
+
+  const filterTodos = (
+    type: FilterTypes,
+  ) => {
+    setFilter(type);
+  };
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
         <header className="todoapp__header">
-          {/* this buttons is active only if there are some active todos */}
           <button type="button" className="todoapp__toggle-all active" />
 
-          {/* Add a todo on form submit */}
-          <form>
+          <form onSubmit={preventSubmit}>
             <input
               type="text"
               className="todoapp__new-todo"
               placeholder="What needs to be done?"
+              value={input}
+              onChange={handleImputTodo}
+              onKeyDown={handleAddTodo}
```

Comment:
It is better to move todo adding on the form submit

Reply:
still not fixed